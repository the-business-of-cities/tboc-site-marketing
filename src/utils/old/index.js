export * from "./bp";
export * from "./colors";
export * from "./shadow";
export * from "./sizing";
export * from "./styles";
export * from "./utility";