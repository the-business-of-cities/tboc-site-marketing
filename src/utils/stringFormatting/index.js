export const formatTelNumber = (num) => {
  if (num.slice(0, 1) === "+" && num.length === 13) {
    return `${num.slice(0, 3)} (0)${num.slice(3, 6)} ${num.slice(
      6,
      9
    )} ${num.slice(9, 13)}`;
  }

  if (num.slice(0, 1) === "0" && num.length === 11) {
    return `${num.slice(0, 4)} ${num.slice(4, 7)} ${num.slice(7, 11)}`;
  } else {
    return num;
  }
};
