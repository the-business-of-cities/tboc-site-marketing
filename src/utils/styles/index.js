export { default as defaultGlobalStyles } from "./defaultGlobalStyles";
export { default as theme } from "./themes/theme";
export * from "./breakpoints";
