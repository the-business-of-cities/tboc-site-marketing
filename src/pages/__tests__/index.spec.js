import React from "react";
import Home from "../index";
import { shallow } from '../../utils/tests';

const data = {
  contentfulHomeSettings: {
    edges: [
      {
        node: {
          homeDescription: {
            homeDescription: 'some description',
          },
          homeBanner: [
            {
              text: 'banner text',
              description: 'something',
              image: {
                file: {
                  url: 'url'
                }
              }
            }
          ]
        }
      }
    ]
  },
  contentfulPage: {
    title: "Home",
    content: [
      {
        backgroundImage: {
          file: {
            url: 'someurl'
          }
        },
        content: "some comtent here",
        ctaTarget: {
          title: 'some title'
        },
        ctaText: 'cta text',
      }
    ],
    partnerCategory: [
      {
        title: 'category name',
        partner: [
          {
            image: 'someurl',
            title: 'partner name',
          }
        ],
      }
    ]
  }
}

describe("Home", () => {
  it("renders correctly", () => {
    expect(shallow(<Home data={data}/>)).toMatchSnapshot();
  });
});
