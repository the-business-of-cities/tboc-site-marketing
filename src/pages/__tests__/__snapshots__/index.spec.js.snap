// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<_default>
  <BlankPage
    banner={
      Object {
        "text": "some description",
      }
    }
    sliderContents={
      Array [
        Object {
          "description": "something",
          "image": Object {
            "file": Object {
              "url": "url",
            },
          },
          "text": "banner text",
        },
      ]
    }
  >
    <Point
      bgImage="someurl"
      cta={
        Object {
          "link": "/some-title",
          "text": "cta text",
        }
      }
      key="0"
      reverse={true}
    />
    <styled.div>
      <styled.div>
        <styled.div
          restrict={true}
        >
          <styled.div>
            <h1>
              A Selection of Our Clients and Partners
            </h1>
            <styled.div>
              <LogoGrid
                key="0"
                logos={
                  Array [
                    Object {
                      "image": "someurl",
                      "link": "partners",
                    },
                  ]
                }
                logosPerRow={
                  Object {
                    "lg": 9,
                    "md": 7,
                    "sm": 5,
                    "xs": 4,
                  }
                }
              />
            </styled.div>
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </BlankPage>
</_default>
`;
